resource-aggregate/cqrs/eventstore/mongodb/getevents.go

```diff
@@ -0,0 +1,99 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/plgd-dev/cloud/resource-aggregate/cqrs/eventstore"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func getEventsQueriesToMongoQuery(groupID string, queries []eventstore.GetEventsQuery) (interface{}, *options.FindOptions) {
+	opts := options.Find()
+	if len(groupID) == 0 {
+		return bson.D{}, opts
+	}
+
+	return nil, opts
+
+	// opts.SetAllowDiskUse(true)
+	// TODO
+	// opts.SetProjection(bson.M{
+	// 	groupIDKey:     1,
+	// 	aggregateIDKey: 1,
+	// 	eventsKey: bson.M{
+	// 		"$filter": bson.M{
+	// 			"input": "$" + eventsKey,
+	// 			"as":    eventsKey,
+	// 			"cond": bson.M{
+	// 				string(signOperator_gte): []string{"$$" + eventsKey + "." + versionKey, "$" + latestSnapshotVersionKey},
+	// 			},
+	// 		},
+	// 	},
+	// })
+	// if len(queries) == 0 {
+	// 	opts.SetHint(groupIDQueryIndex)
+	// 	return bson.D{
+	// 		{Key: groupIDKey, Value: groupID}, {Key: isActiveKey, Value: true},
+	// 	}, opts
+	// }
+
+	// opts.SetHint(groupIDaggregateIDQueryIndex)
+	// orQueries := make([]bson.D, 0, 32)
+	// for _, q := range queries {
+	// 	if q.AggregateID != "" {
+	// 		orQueries = append(orQueries, bson.D{{Key: groupIDKey, Value: groupID}, {Key: aggregateIDKey, Value: q.AggregateID}, {Key: isActiveKey, Value: true}})
+	// 	}
+	// }
+	// return bson.M{
+	// 	"$or": orQueries,
+	// }, opts
+}
+
+func (s *EventStore) getEvents(ctx context.Context, groupID string, queries []eventstore.GetEventsQuery, eventHandler eventstore.Handler) error {
+	filter, opts := getEventsQueriesToMongoQuery(groupID, queries)
+	return s.loadEventsQuery(ctx, eventHandler, nil, filter, opts)
+}
+
+func getNormalizedGetEventsQuery(queries []eventstore.GetEventsQuery) map[string][]eventstore.GetEventsQuery {
+	normalizedQuery := make(map[string][]eventstore.GetEventsQuery)
+	for _, query := range queries {
+		v, ok := normalizedQuery[query.GroupID]
+		if !ok {
+			v = make([]eventstore.GetEventsQuery, 0, 1)
+		}
+		v = append(v, query)
+		normalizedQuery[query.GroupID] = v
```

Comment:
need to be properly implemented

Reply:
Do you mean some changes to this method? Because currently this just splits the mongodb query based on DeviceId (GroupId).

Reply:
When you have Queries: one with AggregateId and GroupId and the second query with the same GroupId, then you absorb the first one.